tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/json/CalendarEventSerializer.scala

```diff
@@ -69,5 +69,20 @@ object CalendarEventSerializer {
   private implicit val calendarEventParseRequestReads: Reads[CalendarEventParseRequest] = Json.reads[CalendarEventParseRequest]
 
   def deserializeCalendarEventParseRequest(input: JsValue): JsResult[CalendarEventParseRequest] = Json.fromJson[CalendarEventParseRequest](input)
-  def serializeCalendarEventResponse(calendarEventResponse: CalendarEventParseResponse): JsValue = Json.toJson(calendarEventResponse)
+  def serializeCalendarEventResponse(calendarEventResponse: CalendarEventParseResponse, properties: Properties): JsValue = {
+    val originalJson = Json.toJson(calendarEventResponse)
+
+    val filteredParsed: JsObject = JsObject(originalJson.transform((__ \ "parsed").json.pick[JsObject])
+      .get
+      .value
+      .map(blobIdToCalendarEvent => blobIdToCalendarEvent._1 -> blobIdToCalendarEvent._2.transform {
+        case jsonObject: JsObject => JsSuccess(properties.filter(jsonObject))
+        case js => JsSuccess(js)
+      }.fold(_ => JsObject(Map(blobIdToCalendarEvent)), o => o)))
+
+    originalJson.transform((__ \ "parsed").json.prune).get
+      .transform((__).json.update(
+        __.read[JsObject].map(prunedParsed => prunedParsed ++ JsObject(Map("parsed" -> filteredParsed)))))
+      .get
+  }
```

Comment:
I was struggling to filter the properties for calendar events here.
Tried something traditional to classic filtering in James but does not work (this case is a bit diff and a bit more complicated):
```
  def serializeCalendarEventResponse(calendarEventResponse: CalendarEventParseResponse, properties: Properties): JsValue =
    Json.toJson(calendarEventResponse)
      .transform((__ \ "parsed").json.update {
        case JsObject(underlying) => {
          val filteredObject: JsObject = JsObject(underlying
            .map(blobIdToCalendarEvent => (blobIdToCalendarEvent._1 -> blobIdToCalendarEvent._2.transform {
              case jsonObject: JsObject => JsSuccess(properties.filter(jsonObject))
              case js => JsSuccess(js)
            }.fold(_ => JsObject(underlying), o => o))))
          val tempJssuccess = JsSuccess(filteredObject)
          tempJssuccess
        }
        case jsValue => JsSuccess(jsValue)
      }).get
```
The `filteredObject` filtered the properties as expected, but the final result is not filtered...

The current solution works for properties filtering but it may need more clean/error handling (some tests failing and I did not have time to fix all of them yet).

Reply:
I'm not sure to get why so complicated to filter properties here... You seem to extract the parsed field then filter it separately and then retake the original json, prune the parsed field and replace with the filtered one... correct?

I don't really have time to put my hands into the code right now but just filtering directly fields in parsed is not possible? How is it because the properties fields are nested in the response?

Reply:
> but just filtering directly fields in parsed is not possible?

Yes, I tried this at first with the code in the comment above. Even with some look with @vttranlina.